connectivity/prox_reporter_makefile/make/build/Makefile

```
@@ -0,0 +1,562 @@
+$(SDKROOT)/platform/core_modules/rf/api \
+$(SDKROOT)/platform/core_modules/rwip/api 
+
+# Add driver includes
+# Top level driver directory
+SDK_DRIVERS := platform/driver
+# Each driver directory has an h file. Get each driver directory
+SDK_DRIVER_DIRS = $(wildcard $(addprefix $(SDKROOT)/, $(SDK_DRIVERS))/*)
+C_INCLUDE_DIRS += $(SDK_DRIVER_DIRS)
+
+C_INCLUDE_DIRS += \
+$(SDKROOT)/platform/include \
+$(SDKROOT)/platform/system_library/include \
+$(SOURCEROOT)/third_party/hash \
+$(SOURCEROOT)/third_party/irng \
+$(SOURCEROOT)/third_party/rand \
+$(SDKROOT)/platform/utilities/otp_cs \
+$(SDKROOT)/platform/utilities/otp_hdr \
+
+# Add User includes
+# Get each directory in source directory
+USER_DIRS := $(wildcard $(addsuffix /, $(APP_SOURCE_DIR))*/) # Have issue when attempt to use APP_SOURCE_DIR directly, using addsuffix as a work around
+# Some, but not all, directories have h files. Get only those with h files
+USER_DIRS_WITH_H_FILE := $(subst //,,$(foreach d,$(USER_DIRS),$(dir $(wildcard  $d/*.h))))
+USER_INCLUDES := $(sort $(USER_DIRS_WITH_H_FILE)) 
+USER_INCLUDES += $(APP_SOURCE_DIR)
+# Add user includes
+C_INCLUDE_DIRS += $(USER_INCLUDES)
+
+##################################################################
+# Sources: .c and .S files 
+
+# The Eclipse project uses virtual folders to organize SDK files into logical groupings. 
+# Here we mimic those groupings with the corresponding "VIRTUAL_" variable. 
+# The order these files are built affects the order they are added to the OBJS variable, which ultimately affects linking order.  
+# We use these groupings to maintain consistency with Eclipse project and produce an identical binary to the Eclipse project
+
+VIRTUAL_SDK_PROFILE_SRCS :=
+VIRTUAL_SDK_DRIVER_SRCS :=
+VIRTUAL_SDK_BOOT_SRCS := 
+VIRTUAL_SDK_BLE_SRCS :=
+VIRTUAL_SDK_ARCH_SRCS :=
+VIRTUAL_SDK_APP_SRCS := 
+
+# sdk_profiles
+# Add additional profiles as necessary. Make sure for any profile you add, you add the corresponding app_module to VIRTUAL_SDK_APP_SUBDIRS below
+VIRTUAL_SDK_PROFILE_SUBDIRS := ble_stack/profiles/bas/bass/src
+VIRTUAL_SDK_PROFILE_SUBDIRS += ble_stack/profiles/dis/diss/src
+VIRTUAL_SDK_PROFILE_SUBDIRS += ble_stack/profiles
+VIRTUAL_SDK_PROFILE_SUBDIRS += ble_stack/profiles/prox/proxr/src
+VIRTUAL_SDK_PROFILE_SUBDIRS += ble_stack/profiles/suota/suotar/src
+# Get all .c files for the sdk_profiles virtrual folder
+SDK_PROFILE_UNSORTED += $(wildcard $(addsuffix /*.c, $(addprefix $(SDKROOT)/, $(VIRTUAL_SDK_PROFILE_SUBDIRS))))
+# SORTING: To maintain the same build/linking order with the Eclipse project build, we need to sort the files based on file name only (diregarding path)
+# Here we remove the path to each file, sort the files, then re-apply to the path to each file maintaining sorted oreder
+SDK_PROFILE_SORTED_WITHOUT_DIR := $(sort $(notdir $(SDK_PROFILE_UNSORTED)))
+SDK_PROFILE_SORTED_BY_FILE_NAME := $(foreach src,$(SDK_PROFILE_SORTED_WITHOUT_DIR),$(filter %$(src),$(SDK_PROFILE_UNSORTED)))
```

Hello! Please consider changing it as follows:

```suggestion
SDK_PROFILE_SORTED_BY_FILE_NAME := $(foreach src,$(SDK_PROFILE_SORTED_WITHOUT_DIR),$(filter %$(src)%,$(SDK_PROFILE_UNSORTED)))
```

Otherwise attempting to compile `platform/core_modules/crypto` fails because `sw_aes.c` and `aes.c` are not properly handled resulting in linking errors.

Applies to all occurrences